### PR TITLE
[DOC] Add `versionadded` and `versionchanged` directives to docstrings

### LINKED ISSUE #489: Start using `versionadded` and `versionupdated` directives in docstrings
## Summary

I would like to use [Sphinx decorators (or directives)](https://deprecated.readthedocs.io/en/latest/sphinx_deco.html) in docstrings to indicate the NiMARE version in which the item was added or last meaningfully changed.

## Additional details

This will help users keep track of changes to NiMARE.

### PR DESCRIPTION
**Pull request by JulioAPeraza**
Closes #489 .

Changes proposed in this pull request:

- Add `versionadded` and `versionchanged` directives to docstrings.

### CODE REVIEW

**Comment by JulioAPeraza** [comment]
@tsalo So far, I have added directives to functions and classes inside annotate. Please, let me know if you would like more detailed comments on `versionchanged` directive.

File: nimare/annotate/utils.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
 """Utility functions for ontology tools."""
 import numpy as np
 import pandas as pd
+from deprecated.sphinx import versionadded
 
 
+@versionadded(version="0.0.3", reason="version < 0.0.3: nimare.annotate.ontology.utils")
```
Comment (tsalo) [inline nimare/annotate/utils.py:7]: I think you'll want to remove these two lines.

**Comment by tsalo** [comment]
This is looking really good, but `build_docs` is failing due to updated dependencies. I think I fixed it in `master` in #502, so you probably just need to update your branch.

File: nimare/decode/continuous.py [hunk 2]
```diff
@@ -220,6 +234,19 @@ def transform(self, img):
...
 
+    .. versionadded:: 0.0.1
+
+    .. versionchanged:: 0.0.2
+        [ENH] Change the order of the first two parameters (img, dataset) -> (dataset, img)
+
+    .. versionchanged:: 0.0.3
+        [ENH] New parameter *feature_group=None*.
+        [ENH] Convert decoder :func:`corr_dist_decode` to class
+        :class:`CorrelationDistributionDecoder`.
```
Comment (tsalo) [inline nimare/decode/continuous.py:245]: It looks like newlines aren't respected, so this renders as a paragraph.

File: nimare/annotate/utils.py [hunk 3]
```diff
@@ -6,6 +6,12 @@
 def _generate_weights(rel_df, weights):
     """Create an IDxID DataFrame linking weight value to each relationship type.
 
+    .. versionadded:: 0.0.2
+
+    .. versionchanged:: 0.0.3
+        [ENH] Submodule was reorginized (`nimare.annotate.ontology.utils` in previous
```
Comment (tsalo) [inline nimare/annotate/utils.py:12]: Reorganized is misspelled as reorginized.

File: nimare/decode/continuous.py (hunk 2 above)
Comment (JulioAPeraza) [reply nimare/decode/continuous.py:245]: We could also use bullet points?
Screen Shot 2021-04-30 at 12 18 56 PM

Comment (tsalo) [reply nimare/decode/continuous.py:245]: That does look much better!

**Comment by JulioAPeraza** [comment]
Here is a list of some possible options:

1. Position in the docstring
a) At the top. 
b) At the bottom.
**Note:** The most common place is at the end, but they mix up with the following function in the docstrings. I think it looks good at the top.

2. `versionchanged`
a) Include changes from all versions.
b) Include changes from the last two releases (or the last two changes).
c) Include when there was a bug.
d) Change of parameters in input and output.
e) Internal enhancement of the function.
**Note:** I agree that `a)` will clutter up the docstrings. Most projects report `versionchanged` for any major change (in the input and output) in the previous 2-3 releases. I included `c)` here just to let the users know that the was a bug in the previous versions and it was fixed in that release.

3. Style
a) Paragraph.
b) Bullet points.
c) Include pull request titles (`FIX`, `ENH`, `REF`).
**Note:** Paragraph looks good too, it is the most common out there. I thought to include `c)` here so the user can easily identify if the change was just an `ENH` or a `FIX`.

I like the combination: 1(a), 2(b, c, d), 3(b, c); but I am probably missing something.

**Comment by tsalo** [comment]
Thank you for describing the problem space. I think 1 depends on how long things end up getting, but I agree that (a) sounds good for now. Regarding the last two, I agree with 2(b, c, d) and 3(b). I'm less sure about 3(c), but if we don't end up liking it we can always remove those prefixes at a later date, so feel free to include the prefixes in this PR.

**Comment by JulioAPeraza** [comment]
I forgot to split option `2b`. 
Should I add `b:` the last two changes (regardless of the version) or `b':` the changes in the last two versions?

**Comment by tsalo** [comment]
I think `b'`, with the caveat that bullets shouldn't necessarily reflect PRs. If there are two PRs addressing related sets of changes, those should be in one bullet, so that the docstring doesn't get too long.

File: nimare/extract/utils.py [hunk 4]
```diff
@@ -153,7 +159,12 @@ def readlinkabs(link):
 
 
 def _download_zipped_file(url, filename=None):
-    """Download from a URL to a file."""
+    """Download from a URL to a file.
+
+    .. versionadded:: 0.0.2
+
+    """
+
```
Comment (tsalo) [inline nimare/extract/utils.py:167]:
```suggestion
    """
```

**Review by tsalo — APPROVED** [review]
The changes LGTM. Thanks!

One caveat: I might adjust some of the notes in later PRs, but I don't have any specific things I want to change at the moment.

**Comment by tsalo** [comment]
Not sure why I didn't merge this last month (sorry about that), but I'm merging now. Thanks for this.

**Pull request merged**